unstructured_ingest/v2/processes/connectors/confluence.py

```diff
@@ -134,54 +133,201 @@ def precheck(self) -> bool:
+
+            Get document permissions. If they exist, they will override space level permissions.
+            Otherwise, apply relevant space permissions (read, administer, delete)
+            """
+
+            # Separate flags to track if view or edit is restricted at the page level
+            page_view_restricted = bool(
+                doc_permissions.get("read", {})
+                .get("restrictions", {})
+                .get("user", {})
+                .get("results")
+                or doc_permissions.get("read", {})
+                .get("restrictions", {})
+                .get("group", {})
+                .get("results")
+            )
+
+            page_edit_restricted = bool(
+                doc_permissions.get("update", {})
+                .get("restrictions", {})
+                .get("user", {})
+                .get("results")
+                or doc_permissions.get("update", {})
+                .get("restrictions", {})
+                .get("group", {})
+                .get("results")
+            )
+
+            normalized_perms_dict = {}
+
+            for action, permissions in doc_permissions.items():
+                restrictions_dict = permissions.get("restrictions", {})
+
+                for entity_type, entity_data in restrictions_dict.items():
+                    for entity in entity_data.get("results"):
+                        if entity_type == "user":
+                            entity_id = entity["accountId"]
+                        elif entity_type == "group":
+                            entity_id = entity["id"]
+
+                        doc_perm_entry = normalized_perms_dict.setdefault(
+                            entity_id,
+                            {
+                                "type": entity_type,
+                                "id": entity_id,
+                                "email_address": entity.get("email", None),
+                                "roles": set(),
+                            },
+                        )
+                        doc_perm_entry["roles"].add(action)
+                        # edit permission implies view permission
+                        if action == "update":
+                            doc_perm_entry["roles"].add("read")
+
+            for space_perm in space_permissions:
+                space_operation = space_perm["operation"]["key"]
+                space_target_type = space_perm["operation"]["targetType"]
+                space_entity_id = space_perm["principal"]["id"]
+                space_entity_type = space_perm["principal"]["type"]
+
+                if space_target_type == "space":
+                    if space_operation == "read" and not page_view_restricted:
+                        # Apply space-level view permissions if no page restrictions exist
+                        perm_entry = normalized_perms_dict.setdefault(
+                            space_entity_id,
+                            {
```

Comment:
Can we concretely define the unified format we want to store permissions in somewhere outside of this module?